runtime-sdk/src/modules/core/types.rs

```diff
@@ -32,3 +32,37 @@ pub struct CallDataPublicKeyQueryResponse {
     /// Public key used for deriving the shared secret for encrypting call data.
     pub public_key: SignedPublicKey,
 }
+
+#[derive(Debug, Copy, Clone, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum MethodHandlerKind {
+    Call,
+    // `Prefetch` is omitted because it is an implementation detail of handling `Call`s.
+    Query,
+    MessageResult,
+}
+
+#[derive(Debug, Clone, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct MethodHandlerInfo {
+    pub kind: MethodHandlerKind,
+    pub name: String,
+}
+
+/// Metadata for an individual module.
+#[derive(Clone, Debug, cbor::Encode, cbor::Decode)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct ModuleInfo {
+    pub version: u32,
+    pub params: Vec<u8>,
```

Comment:
Should this be of type `cbor::Value` instead? That would make the parameters always be valid CBOR embedded into the struct instead of having this arbitrary byte array. On the deserialization/Go side you can just use `cbor.RawMessage`.

Reply:
Good call. And on the TS side we're more flexible so we can have `any`, and it deserializes nicely. Updated the type and tests.